src/msg/async/rdma/RDMAConnectedSocketImpl.cc

```diff
@@ -233,18 +239,19 @@ void RDMAConnectedSocketImpl::handle_connection() {
         ldout(cct, 10) << __func__ << " server is already active." << dendl;
         return ;
       }
+      r = activate();
+      assert(!r);
       r = infiniband->send_msg(cct, tcp_fd, my_msg);
       if (r < 0) {
         ldout(cct, 1) << __func__ << " server ack failed." << dendl;
         dispatcher->perf_logger->inc(l_msgr_rdma_handshake_errors);
         fault();
         return ;
       }
-      r = activate();
-      assert(!r);
     } else { // ack from client
       connected = 1;
-      cleanup();
+      ldout(cct, 10) << __func__ << " handshake of rdma is done. server connected: " << connected << dendl;
+      //cleanup();
```

Comment:
I have confirmed locally that it can fix the CLOSE_WAIT.
I'm still confused about that if we close the tcp_fd after RDMA connection established, how do RDMA connection detect that other side are not online when there is no data to exchange?